reference/vane-apis/iris.md

```diff
@@ -0,0 +1,179 @@
++++
+title = "Iris Public API"
+weight = 5
+template = "doc.html"
++++
+
+# Iris
+
+In this document we describe the public interface for Iris. Namely, we describe
+each `task` that Iris can be `%pass`ed, and which `gift`(s) Iris can `%give` in return.
+
+## Tasks
+
+### `%born`
+
+Each time you start your Urbit the Arvo kernel calls the `%born` `task` for
+Iris. This causes Iris to cancel any outstanding http requests and resets all
+connection states.
+
+#### Accepts
+
+```hoon
+[ ~ ]
+```
+
+#### Returns
+
+In response to a `%born` `task` Iris `give`s a `%http-response %cancel` `gift`
+to each outstanding connection.
+
+
+### `%cancel-request`
+
+This `task` cancels a previous fetch. Iris knows which request is meant based on the
+`duct` that the `task` comes on.
+
+ We note that `%cancel-request` is also a `gift` that Iris can `%give`. 
+
+#### Accepts
+
+```hoon
+[ ~ ]
+```
+
+#### Returns
+
+```hoon
+[%cancel-request id=@ud]
+```
+Iris `%give`s a `%cancel-request` `gift` in response to this `task`. `id` is
+obtained via a `(map duct @ud)`, with the `duct` corresponding to one along
+which the `%cancel-request` `task` came with. 
```

Comment:
It'd be worth noting the `%cancel-request` it gives is to Unix.  It's canceling the request in the runtime.

Reply:
I think I see why this is but I want to make sure - the `%born` `task` sets `outbound-duct.state.ax` to `duct`, and `%born` originates from Unix so `outbound-duct` always points to Unix. Then `outbound-duct` is the `duct` on which `+cancel` returns, thus Iris gives `%cancel-request` to Unix.

Reply:
Yes, precisely

Reply:
Is this return really a response to the `%cancel-request` `task` then since it does not return on the same `duct` that it was passed with? Or is it considered a response to `%born` somehow?

There is this comment in `iris.hoon` that makes it sound like that this is really a return of `%born`, but its not an ideal state of affairs and should really be a pass t unix.

```
:
    ::  the original eyre keeps track of the duct on %born and then sends a
    ::  %give on that duct. this seems like a weird inversion of
    ::  responsibility, where we should instead be doing a pass to unix. the
    ::  reason we need to manually build ids is because we aren't using the
    ::  built in duct system.
    ::
    ::  email discussions make it sound like fixing that might be hard, so
    ::  maybe i should just live with the way it is now?
    ::
```

Reply:
I wouldn't call it a "response" per-se, but it's triggered by the `%cancel-request` task.  Much of what Iris does is just translating from other vanes to unix, so most things you ask Iris to do just turn into a card destined for Unix.

Reply:
Here's some changes to reflect this. I added a note at the top about "returns to unix" to avoid repeating the same information for `%request` and `%cancel-request`, but I could also see the merit of putting that information under one of the `task`s and then referring to it from the other one.

https://github.com/urbit/docs/pull/837/commits/f23b595b63a162e815eb67d1e1e0555c05c315b2